drift/lib/src/sentry_query_executor.dart

```diff
@@ -0,0 +1,170 @@
+import 'dart:async';
+
+import 'package:drift/backends.dart';
+import 'package:drift/drift.dart';
+import 'package:meta/meta.dart';
+import 'package:sentry/sentry.dart';
+import 'sentry_span_helper.dart';
+import 'sentry_transaction_executor.dart';
+
+/// Signature of a function that opens a database connection when instructed to.
+typedef DatabaseOpener = FutureOr<QueryExecutor> Function();
+
+/// The Sentry Query Executor.
+///
+/// If the constructor parameter queryExecutor is not null, it will be used
+/// instead of the default [LazyDatabase].
+@experimental
+class SentryQueryExecutor extends QueryExecutor {
+  Hub _hub;
+
+  final _spanHelper = SentrySpanHelper(
+    // ignore: invalid_use_of_internal_member
+    SentryTraceOrigins.autoDbDriftQueryExecutor,
+  );
+
+  final QueryExecutor _queryExecutor;
+
+  final String _dbName;
+
+  @internal
+  // ignore: public_member_api_docs
+  static const dbNameKey = 'db.name';
+
+  @internal
+  // ignore: public_member_api_docs
+  static const dbOp = 'db';
+
+  @internal
+  // ignore: public_member_api_docs
+  static const dbSystemKey = 'db.system';
+
+  @internal
+  // ignore: public_member_api_docs
+  static const dbSystem = 'sqlite';
```

Comment:
yup, https://drift.simonbinder.eu/faq/#can-i-view-a-drift-database

since drift itself is not a database but uses sqlite under the hood, I think sqlite should be alright, we also use sqlite for sqflite.